### PR TITLE
SOC-2144 | Use canLock, canUnlock, isLocked for (un)locking actions

### PR DESCRIPTION
**Pull request by rafalkalinski**
## Links

https://wikia-inc.atlassian.net/browse/SOC-2144

## Description

Use canLock, canUnlock, isLocked for (un)locking actions

## Reviewers

@kvas-damian, @bkoval

### CODE REVIEW

File: front/main/app/components/discussion-more-options.js [hunk 1]
```diff
@@ -21,12 +21,12 @@ export default Ember.Component.extend({
 			!this.get('post.isDeleted');
 	}),
 
-	canLock: Ember.computed('isLockable', 'post.isLocked', 'post.userData.permissions.canDelete', function () {
+	canLock: Ember.computed('isLockable', 'post.isLocked', 'post.userData.permissions.canLock', function () {
 		// @ToDo use canLock for this -> SOC-2144
```
Comment (kvas-damian) [inline front/main/app/components/discussion-more-options.js]: Please remove TODO

**Comment by kvas-damian** [comment]
LGTM

**Comment by Bloodmoon24** [comment]
LGTM

File: front/main/app/components/discussion-more-options.js [hunk 2]
```diff
@@ -21,12 +21,11 @@ export default Ember.Component.extend({
 			!this.get('post.isDeleted');
 	}),
 
-	canLock: Ember.computed('isLockable', 'post.isLocked', 'post.userData.permissions.canDelete', function () {
-		// @ToDo use canLock for this -> SOC-2144
-		return this.get('isLockable') && !this.get('post.isLocked') && this.get('post.userData.permissions.canDelete');
+	canLock: Ember.computed('isLockable', 'post.isLocked', 'post.userData.permissions.canLock', function () {
```
Comment (hakubo) [inline front/main/app/components/discussion-more-options.js]: you could add one more computed property

`postIsOpen: Ember.computed.not('post.isLocked),`

and then you could use built in computed property and
`canLock: Ember.computed.and('isLockable', 'postIsOpen', 'post.userData.permissions.canLock');`

what do you think?

**Comment by hakubo** [comment]
one minor comment, lgtm

File: front/main/app/components/discussion-more-options.js (hunk 2 above)
Comment (rafalkalinski) [inline front/main/app/components/discussion-more-options.js]: We would not use postIsOpen property for anything else. Does it have any other advantages except the readability?

**Pull request merged**